### ISSUE TITLE
ENH: automatically parse units (Pluto)

### ISSUE THREAD

**Issue by xshaokun** [neutrinoceros/yt_idefix#54]
Hi,

Coming from yt-project/yt#3711

> Sound like you want to load as
> ```python
> ds = yt_idefix.load("your_file.vtk", unit_system="code")
> ```

The key `code` doesn't work as other yt frontends do. I checked the code and found that you have not implemented unit conversion in `IndefixDataset._set_code_unit_attributes` yet. I am willing to make it in my spare time, in the way similar to my own package but using `unyt` here. For doing so, I needed to add a new option for specifying the location of 'definition.h' file.

I don't know about `idefix`. Are there any different conventions compared to `PLUTO`?

**Comment by neutrinoceros** [comment]
In Idefix we never define these normalization factors. If they can be obtained from a header file in Pluto then it would make sense to have a specialized dataset class `PlutoVtkDataset`, that could inherit from `IdefixVtkDataset`.

Btw just curious, can you actually load Pluto vtk files without changing ˋIdefixVtkDataset._is_valid` ?

**Comment by xshaokun** [comment]
> Btw just curious, can you actually load Pluto vtk files without changing ˋIdefixVtkDataset._is_valid` ?

Yes, only with a few warnings as follows:

``` IPython
In [4]: ds = yt_idefix.load("data.0004.vtk", geometry='cartesian')
/share/home/skxie/Apps/anaconda3/lib/python3.8/site-packages/yt_idefix/data_structures.py:204: UserWarning: Could not determine Idefix version from file header '# vtk DataFile Version 2.0\nPLUTO 4.4-patch2 VTK Data\n'
  warnings.warn(
yt : [INFO     ] 2021-12-09 19:40:36,651 Parameters: current_time              = -1
yt : [INFO     ] 2021-12-09 19:40:36,651 Parameters: domain_dimensions         = [200 200 300]
yt : [INFO     ] 2021-12-09 19:40:36,651 Parameters: domain_left_edge          = [-1.  -1.  -0.2]
yt : [INFO     ] 2021-12-09 19:40:36,651 Parameters: domain_right_edge         = [1.         1.         2.79999995]
yt : [INFO     ] 2021-12-09 19:40:36,651 Parameters: cosmological_simulation   = 0

In [5]: ds.print_stats()
level	# grids	       # cells	     # cells^3
----------------------------------------------
  1	     1	      12000000	           229
----------------------------------------------
   	     1	      12000000


t = -1.00000000e+00 = -1.00000000e+00 s = -3.16880878e-08 years

Smallest Cell:
	Width: 3.241e-27 Mpc Mpc
	Width: 3.241e-21 pc pc
	Width: 6.685e-16 AU AU
	Width: 1.000e-02 cm cm

In [6]: ds.field_list
Out[6]:
[('idefix-vtk', 'phi'),
 ('idefix-vtk', 'prs'),
 ('idefix-vtk', 'rho'),
 ('idefix-vtk', 'tmp'),
 ('idefix-vtk', 'tr1'),
 ('idefix-vtk', 'vx1'),
 ('idefix-vtk', 'vx2'),
 ('idefix-vtk', 'vx3')]
```


I just realized that it should not to work because there's no "Idefix" string in my vtk files.

**Comment by neutrinoceros** [comment]
> I just realized that it should not to work because there's no "Idefix" string in my vtk files.

My point exactly. So this looks like a bug to me.
Could you share a Pluto vtk file (preferably a small one) that I could use for testing ? I'll start the process of adding a specialized class for Pluto to maintain compatibility and release a new version with the fix :)

**Comment by xshaokun** [comment]
That's great! Here is a Pluto vtk data of 3d Sodov test problem given by Pluto:

transfer link: https://cowtransfer.com/s/58a7ef00295645 or access [ cowtransfer.com ] and input extract code: 8kocic to download;

The link is only valid within one day.

**Comment by neutrinoceros** [comment]
I just released yt_idefix 0.8 which has more explicit support for pluto data. Units are not being parsed yet, feel free to open a PR for this whenever is most convenient to you. If you need any additional help, feel free to open a _draft_ PR with whatever progress you're able to make on your own, even if it doesn't work !

**Comment by neutrinoceros** [comment]
triaging this as "help wanted" to advertise that I'm not doing this work myself, but @xshaokun is on his way to solve it !

**Comment by neutrinoceros** [comment]
closed with #57
